### PR TITLE
Add instruction on how to handle the potential linker error on Linux

### PR DESCRIPTION
**Pull request by skyline75489**
The original issue is #16683, which contains a https://github.com/pytorch/pytorch/issues/16683#issuecomment-459982988 that suggests manually un-shadowing the `ld`. 

A better approach can be found at https://github.com/ContinuumIO/anaconda-issues/issues/11152#issuecomment-573120962, which suggests that using a newer version can effectively fix this. 

It took me quite some time to realize that this is in fact an issue caused by Anaconda. I think we should add it in README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -149,7 +149,7 @@ They require JetPack 4.2 and above, and [@dusty-nv](https://github.com/dusty-nv)
 
 ### From Source
 
-If you are installing from source, you will need Python 3.6 or later and a C++14 compiler. Also, we highly recommend installing an [Anaconda](https://www.anaconda.com/distribution/#download-section) environment.
+If you are installing from source, you will need Python 3.6.2 or later and a C++14 compiler. Also, we highly recommend installing an [Anaconda](https://www.anaconda.com/distribution/#download-section) environment.
```
Comment (skyline75489) [inline README.md:152]: The minimum version supported is `3.6.2` in `setup.py`.

File: README.md [hunk 2]
```diff
@@ -207,6 +207,16 @@ export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
...
+
+```plaintext
+build/temp.linux-x86_64-3.7/torch/csrc/stub.o: file not recognized: file format not recognized
+collect2: error: ld returned 1 exit status
+error: command 'g++' failed with exit status 1
+```
+
+This is caused by `ld` from Conda environment shaddowing the system `ld`. You should use a newer version of Python that fixes this issue. The recommended Python version is 3.6.10+, 3.7.6+ and 3.8.1+.
+
 On macOS
```
Comment (skyline75489) [inline README.md:220]: I don't know if this also happens on macOS.  Based on my experience and the original issue, this is a reproducible issue on Linux.

Windows is spared from this issue.

**Review by heitorschueroff — CHANGES_REQUESTED** [review]
Thanks for your contribution. Just a small typo fix and it will be good.

File: README.md [hunk 3]
```diff
@@ -207,6 +207,16 @@ export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
...
 
+Note that if you are using [Anaconda](https://www.anaconda.com/distribution/#download-section), you may experience an error caused by the linker:
+
+```plaintext
+build/temp.linux-x86_64-3.7/torch/csrc/stub.o: file not recognized: file format not recognized
+collect2: error: ld returned 1 exit status
+error: command 'g++' failed with exit status 1
+```
+
+This is caused by `ld` from Conda environment shaddowing the system `ld`. You should use a newer version of Python that fixes this issue. The recommended Python version is 3.6.10+, 3.7.6+ and 3.8.1+.
```
[the remark this answers is not in the document]
Comment (skyline75489) [reply README.md]: Thanks for the prompt reply. Fixed.

File: README.md (hunk 1 above)
Comment (skyline75489) [reply README.md:152]: This actually will cause build failure on current master with Python < 3.6.2, due to the usage of [typing.NoReturn](https://docs.python.org/3/library/typing.html#typing.NoReturn) which was first introduced in 3.6.2

**Review by heitorschueroff — APPROVED** [review]

**Pull request closed without merging**